pkg/engine/utils.go

```diff
@@ -282,10 +282,27 @@ func matchSubjects(ruleSubjects []rbacv1.Subject, userInfo authenticationv1.User
 }
 
 // MatchesResourceDescription checks if the resource matches resource description of the rule or not
-func MatchesResourceDescription(resourceRef unstructured.Unstructured, ruleRef kyvernov1.Rule, admissionInfoRef kyvernov1beta1.RequestInfo, dynamicConfig []string, namespaceLabels map[string]string, policyNamespace string) error {
+func MatchesResourceDescription(resourceRef unstructured.Unstructured, ctx *PolicyContext, ruleRef kyvernov1.Rule, policyNamespace string) error {
 	rule := ruleRef.DeepCopy()
+
+	var dynamicConfig []string
+	if len(ctx.excludeGroupRole) > 0 {
+		dynamicConfig = ctx.excludeGroupRole
+	}
+
+	namespaceLabels := ctx.namespaceLabels
+	if ctx.PolicyExceptions != nil && !ctx.PolicyExceptions.IsNil() {
```

Comment:
@JimBugwadia Can you help with a review on the nil check on this line?
I've made some updates based our last pair, but I'm not sure if I'm quite following with your suggestion.